docs/source/API/extension_dev.rst

```diff
@@ -0,0 +1,193 @@
+        'asreview.models.query': [
+            # define query_strategy algorithms
+        ]
+    },
+
+This code registers the model with name ``example``.
+
+.. |template_link| raw:: html
+
+    <a href="https://github.com/asreview/template-extension-new-model"
+    target="_blank"> template for extending ASReview</a>
+
+Subcommand extensions 
+---------------------
+An extension of the :class:`asreview.entry_points.base.BaseEntryPoint` class.
+
+Subcommand extensions are programs that create a new entry point for ASReview.
+From this entry point the Python API can be used in many ways (like ``plot`` or
+``simulate``).
+
+Extensions in ASReview are Python packages and can extend the
+subcommands of asreview (see ``asreview -h``).
+
+The easiest way to create a new subcommand is by defining a class that can be
+used as a new entry point for ASReview. This class should inherit from
+:class:`asreview.entry_points.base.BaseEntryPoint`. Add the functionality to the
+class method ``execute``.
+
+.. code:: python
+
+    from asreview.entry_points import BaseEntryPoint
+
+    class ExampleEntryPoint(BaseEntryPoint):
+
+        description = "Description of example extension"
+        extension_name = "asreview-example"  # Name of the extension
+        version = "1.0"  # Version of the extension in x.y(.z) format.
+
+        def execute(self, argv):
+            pass  # Implement your functionality here.
+
+It is strongly recommended to define the attributes ``description``,
+``extension_name``, and ``version``.
+
+The class method ``execute`` accepts a positional arugument (``argv`` in this
+example).  First create the functionality you would like to be able to use in
+any directory. The argument ``argv`` are the command line arguments left after
+removing asreview and the entry point.
+
+It is advised to place the newly defined class ``ExampleEntryPoints`` in the
+following package structure:
+``asreviewcontrib.{extension_name}.{your_modules}``. For example:
+
+.. code:: bash
+
+    ├── README.md
+    ├── asreviewcontrib
+    │   └── example
+    │       ├── __init__.py
+    │       ├── entrypoint.py
+    │       └── example_utils.py
+    ├── setup.py
+    └── tests
+
+
+Create a ``setup.py`` in
+the root of the package, and set the keyword argument `entry_points` of
+``setup()`` under ``asreview.entry_points``, for example:
+
+.. code:: python
+
+    entry_points={
+        "asreview.entry_points": [
+            "example = asreviewcontrib.example.entrypoint:ExampleEntryPoint",
+        ]
+    }
+
+After installing this package, ASReview is extended with the ``asreview
+example`` subcommand. See ``asreview -h`` for this option.
+
+An example of a subcommand extension is the `Visualization Extension
+<https://github.com/asreview/asreview-visualization>`_
+
+Dataset extensions
+------------------
+An extension of the :class:`asreview.datasets.BaseDataSet` class.
+
+Dataset extensions integrate new datasets for use in ASReview. Adding datasets
```

Comment:
Can you maybe add an explanation how the 'dataset' as used here, is different from a dataset you can import or a benchmark dataset you can see in the exploration mode?